mlrun/runtimes/sparkjob/abstract.py

```diff
@@ -451,16 +462,46 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
             update_in(job, "spec.mainApplicationFile", self.spec.command)
 
         verify_list_and_update_in(job, "spec.arguments", self.spec.args or [], str)
-        self._submit_job(job, meta.namespace)
+        self._submit_job(job, meta.namespace, code)
 
         return None
 
     def _enrich_job(self, job):
         raise NotImplementedError()
 
-    def _submit_job(self, job, namespace=None):
+    def _submit_job(self, job, namespace=None, code=None):
         k8s = self._get_k8s()
         namespace = k8s.resolve_namespace(namespace)
+        if code:
+            k8s_config_map_name = f"{self.metadata.name}-{uuid.uuid4().hex[:8]}"
+            k8s_config_map = client.V1ConfigMap()
+            k8s_config_map.metadata = client.V1ObjectMeta(
+                name=k8s_config_map_name,
+                namespace=namespace,
+                labels=get_in(job, "metadata.labels"),
+            )
```

Comment:
In `_run` you already have `meta = self._get_meta(runobj, True)` no need to regenerate it, just pass it here